src/site/_includes/components/cookie/index.ts

```diff
@@ -0,0 +1,74 @@
+import { getCookie, setCookie } from './cookie'
+
+const COOKIE_BANNER_CLASS_PREFIX = 'js-cookie-guidance-banner'
+const COOKIE_PREFERENCE_KEY = 'cookie_preference'
+
+const getClassName = (className: string) => `.${COOKIE_BANNER_CLASS_PREFIX}--${className}`
+
+type CookieMessageToggles = {
+  show: VoidFunction
+  hide: VoidFunction
+}
+
+const getCookieMessageElements = <T extends readonly string[]>(
+  parentElement: Element,
+  types: T,
+): Record<T[number], CookieMessageToggles> => (
+  types.reduce((acc, type) => {
+    const element = parentElement.querySelector<HTMLDivElement>(getClassName(type))
+    if (!element) {
+      throw new Error('Message not found')
+    }
+    acc[type] = {
+      show: () => {
+        element.removeAttribute('hidden')
+        types.forEach((currentType) => {
+          if (currentType !== type) {
+            acc[currentType].hide()
+          }
+        })
+      },
+      hide: () => {
+        element.setAttribute('hidden', 'true')
+      },
+    }
+    return acc
+  }, {} as Record<string, CookieMessageToggles>)
+)
+
+export function initCookieBanner() {
+  const cookieBannerEl = document.querySelector(getClassName('container'))
+
+  const cookiePreference = getCookie(COOKIE_PREFERENCE_KEY)
+  if (typeof cookiePreference === 'string' || !cookieBannerEl) {
+    return
+  }
+  const messages = ['permission', 'accept', 'reject'] as const
+
+  const { permission, accept, reject } = getCookieMessageElements<typeof messages>(cookieBannerEl, messages)
+
+  cookieBannerEl.addEventListener('click', (event) => {
+    if (!(event.target instanceof HTMLButtonElement)) {
+      return
+    }
+    const button = event.target
+    switch (button.value) {
+      case 'accept':
+        accept.show()
+        setCookie(COOKIE_PREFERENCE_KEY, 'true', 365)
+        break
+      case 'reject':
+        setCookie(COOKIE_PREFERENCE_KEY, 'false', 365)
+        reject.show()
+        break
+      case 'hide':
+        cookieBannerEl.setAttribute('hidden', 'true')
```

Comment:
[Should the value be `hidden`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden#description):
```suggestion
        cookieBannerEl.setAttribute('hidden', 'hidden')
```